### DISCUSSION TITLE
LESSON 6 CAST-CALL ERROR

### DISCUSSION THREAD

**Discussion by Dzgndgn** [Cyfrin/foundry-full-course-cu#1355] [Q&A]
Althought I deployed my contract I take error when I take error.
```bash
cast call 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 "retrieve()(uint256)" --rpc-url $RPC_URL
Error: 
contract 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 does not have any code
```

**Comment by ghost** [comment]
Two options to consider:

A. Maybe you're querying the contract on the incorrect network so make sure you're using the correct RPC url
B. If you deployed on anvil then the local node needs to be running

**Reply by AbigirlMuchineripi** [reply]
hie @Dzgndgn , you need to use the contract address from run-latest.json file , check the screenshot below. I was stuck too but worked for. Kindly inform if you solved this.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/94177156/c367b9c7-83d8-47cf-b733-e63610f262dc)

**Comment by ighodaro-daniel** [comment]
greetings my fellow Dev, 
i am trying to use the terminal to interact with my contract  by sending a transaction  using the CAST command 
```cast send 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb9226 "store(uint256)" 123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY
```` 
but the terminal keeps printing this error, 
```
Error: 
Failed to get resolver from the ENS registry: buffer overrun while deserializing
``` 
pls how do I scale through this error

**Reply by AbigirlMuchineripi** [reply]
hello @ighodaro-daniel ,can you provide your code.

**Reply by ighodaro-daniel** [reply]
here is the code  
''' //SPDX-License-Identifier: MIT
pragma solidity ^0.8.18; // solidity version 0.8.19

contract SimpleStorage {
    uint256 public favoriteNumber;

    function store(uint256 updateFavoriteNum) public virtual {
        favoriteNumber = updateFavoriteNum;
    }

    struct Users {
        uint256 number;
        string name;
    }
    mapping(string => uint256) public getNumberByName;

    function retrieval() public view returns (uint256) {
        return favoriteNumber;
    }

    Users[] public listOfPersons;

    function addPeople(string memory name, uint256 number) public {
        listOfPersons.push(Users(number, name));
        getNumberByName[name] = number;
    }
}
'''  
after much troubleshooting , i realised that it was the contract address provided by anvil that was the problem , so i had to change it and it worked  but i discover another error , i  am trying to view the updated variable calling the function that hold it with the command
  cast call 0x70997970C51812dc3A010C7d01b50e0d17dc79C8 "retrieval()"  
 the terminal keeps giving an empty variable
```terminal response
0x
```
what could be the problem?

**Reply by AbigirlMuchineripi** [reply]
Are you not making an error by writing the "retrieval()" instead of retrieve()? Kindly recheck and tell me if it worked.

**Reply by ighodaro-daniel** [reply]
nope it's the signal function is retrieval() and not retrieve(), i posted the full code
